service/registry/src/main/java/io/helidon/service/registry/CoreServiceRegistry.java

```diff
@@ -319,11 +326,14 @@ public List<ServiceInfo> lookupServices(Lookup lookup) {
             // table scan :-(
             metrics.fullScan();
             // we need to go through each service descriptor if it matches
-            servicesByDescriptor.keySet()
+            var tmpResult = servicesByDescriptor.keySet()
                     .stream()
                     .filter(lookup::matches)
                     .sorted(SERVICE_INFO_COMPARATOR)
-                    .forEach(result::add);
+                    .collect(Collectors.toUnmodifiableList());
```

Comment:
Not exactly a problem I think, but would not `toList()` be nicer here? Or is there a reason for this approach?

Reply:
`.toList()` instead of `.collect(Collectors.toUnmodifiableList())` would not be a problem, just did not know about that method...